### DISCUSSION TITLE
plus signs regex quantifiers

### DISCUSSION THREAD

**Discussion by ronaldpk** [symfony/symfony#48995] [Q&A] [answered]
In the `getMethod` method of the HTTPFoundation Request class there's this line:

 `if (!preg_match('/^[A-Z]++$/D', $method)) {`

`$method` is supposed to match simple strings like PUT, POST or DELETE.  I'm just wondering why there are two plus signs in the pattern. Or is it a typo?

**Comment by MatTheCat** [comment, marked as answer]
`++` is a [possessive quantifier](https://www.regular-expressions.info/possessive.html) and it is used in the case of a method override, so the method could be anything here.
